core/domain/learner_progress_services.py

```diff
@@ -445,6 +447,61 @@ def mark_collection_as_incomplete(user_id, collection_id):
         _save_incomplete_activities(incomplete_activities)
 
 
+def add_topic_to_learn(user_id, topic_id):
```

Comment:
Similarly, maybe rename to `validate_and_add_topic_to_learn_goal`? Since currently add_topic_to_learn and mark_topic_to_learn seems to do the same thing, name-wise.

Reply:
Done.